tests/integration-service/namespace-backed-environments.go

```diff
@@ -260,6 +262,109 @@ var _ = framework.IntegrationServiceSuiteDescribe("Namespace-backed Environment
+			Expect(metadata.GetLabelsWithPrefix(updatedSnapshot, snapshotRerunLabel)).NotTo(BeEmpty())
+		})
+
+		When("An snapshot is updated with a rerun label for a given scenario", func() {
+			It("checks if the rerun label was removed from the Snapshot", func() {
+				Eventually(func() error {
+					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot(snapshot.Name, "", "", testNamespace)
+					if err != nil {
+						return fmt.Errorf("encountered errror while getting Snapshot %s/%s: %w", snapshot.Name, snapshot.Namespace, err)
+					}
+
+					if metadata.HasLabel(snapshot, snapshotRerunLabel) {
+						return fmt.Errorf("the Snapshot %s/%s shouldn't contain the %s label", snapshot.Name, snapshot.Namespace, snapshotRerunLabel)
+					}
+					return nil
+				}, time.Minute*2, time.Second*5).Should(Succeed())
+			})
+
+			It("creates an Ephemeral Environment", func() {
+				Eventually(func() error {
+					ephemeralEnvironment, err = f.AsKubeAdmin.GitOpsController.GetEphemeralEnvironment(snapshot.Spec.Application, snapshot.Name, newIntegrationTestScenario.Name, testNamespace)
+					return err
+				}, time.Minute*3, time.Second*1).Should(Succeed(), fmt.Sprintf("timed out when waiting for the creation of Ephemeral Environment related to snapshot %s", snapshot.Name))
+				Expect(err).ToNot(HaveOccurred())
+				Expect(ephemeralEnvironment.Name).ToNot(BeEmpty())
+			})
+
+			It("checks for SEB after Ephemeral env has been created", func() {
+				seb, err = f.AsKubeAdmin.CommonController.GetSnapshotEnvironmentBinding(applicationName, testNamespace, ephemeralEnvironment)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(seb).ToNot(BeNil())
+				Expect(seb.Spec.Snapshot).To(Equal(snapshot.Name))
+				Expect(seb.Spec.Application).To(Equal(applicationName))
+				Expect(seb.Spec.Environment).To(Equal(ephemeralEnvironment.Name))
+				Expect(seb.Spec.Components).ToNot(BeEmpty())
+			})
+
+			It("checks if the new integration pipelineRun started", Label("slow"), func() {
+				reRunPipelineRun, err := f.AsKubeDeveloper.IntegrationController.WaitForIntegrationPipelineToGetStarted(newIntegrationTestScenario.Name, snapshot.Name, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(reRunPipelineRun).ShouldNot(BeNil())
+
+				Expect(reRunPipelineRun.Labels[snapshotAnnotation]).To(ContainSubstring(snapshot.Name))
+				Expect(reRunPipelineRun.Labels[scenarioAnnotation]).To(ContainSubstring(newIntegrationTestScenario.Name))
+				Expect(reRunPipelineRun.Labels[environmentLabel]).To(ContainSubstring(ephemeralEnvironment.Name))
+			})
+
+			It("checks if all integration pipelineRuns finished successfully", Label("slow"), func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForAllIntegrationPipelinesToBeFinished(testNamespace, applicationName, snapshot)).To(Succeed())
+			})
+
+			It("checks if the status of the re-triggered integration test is reported in the Snapshot", FlakeAttempts(3), func() {
```

Comment:
since we're checking the name of PLR instead:
```suggestion
			It("checks if the name of the re-triggered pipelinerun is reported in the Snapshot", FlakeAttempts(3), func() {
```
wdyt?

Reply:
ack, updated.